File: design-patterns/go/visitor/ui_element/list_element.go
// list_element.go

package main

type ListElement struct {
	lines []string
}

func NewListElement(lines []string) (listElement *ListElement) {
	listElement = &ListElement{}
	listElement.lines = lines
	return
}

func (listElement *ListElement) GetListItems() []string {
	return listElement.lines
}

func (listElement *ListElement) AppendElement(visitor Visitor) {
	visitor.AppendList(listElement)
}

File: design-patterns/php/visitor/uielement/ListElement.php
<?php
// ListElement.php

namespace BigBoxCode\DesignPattern\Visitor\UiElement;

class ListElement implements UIElement {
    private array $lines;

    public function __construct(array $lines) {
        $this->lines = $lines;
    }

    public function getListItems(): array {
        return $this->lines;
    }

    public function appendElement(Visitor $visitor): void {
        $visitor->appendList($this);
    }
}